### PR TITLE
hiveplot

### PR DESCRIPTION
**Pull request by ericmjl**

### CODE REVIEW

**Comment by patricksnape** [comment]
@jakirkham This recipe is a bit atypical in that it's a `git mv` of the example recipe. Interestingly, however, this fails on Appveyor only because on Windows we try to remove the example recipe and the script is not resilient to it being missing (clearly in contrast Unix). I imagine we would change this recipe to not move the example, but still, perhaps we should fix that small oversight too?

**Comment by ericmjl** [comment]
@patricksnape now that I understand how it's failing, I can put back the example recipe in place.

**Comment by jakirkham** [comment]
> I imagine we would change this recipe to not move the example, but still, perhaps we should fix that small oversight too?

Which oversight, @patricksnape? The one that causes Unix not to fail or the one that causes Windows to fail? It seems the failure was helpful for catching this issue. So, maybe the former?

File: recipes/hiveplot/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    # Remember to specify the license variants for BSD, Apache, GPL, and LGLP.
+    # Prefer the short version, e.g: GPL-2.0 instead of GNU General Public License version 2.0
+    # See https://opensource.org/licenses/alphabetical
+    license: MIT
+    summary: 'Hive plots in Python!'
+
+extra:
+    recipe-maintainers:
+        # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
+        - {{ maintainer }}
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: @ericmjl, please add yourself as a maintainer. 😄

File: recipes/hiveplot/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        - setuptools
+        - matplotlib
+        - numpy
+        - pip
+        - icu
+    run:
+        - python
+        - matplotlib
+        - numpy
+        - pip
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: I'm guessing `pip` is not a run-time requirement, right?

File: recipes/hiveplot/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        - icu
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Why do we need `icu`?

File: recipes/hiveplot/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+build:
+    number: 0
+    # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+    script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+    build:
+        - python
+        # When setuptools is available add the `--single-version-externally-managed --record record.txt` above.
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Please drop this comment.

File: recipes/hiveplot/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+source:
+    fn: {{ pkgname }}-{{ version }}.tar.gz
+    url: https://pypi.io/packages/source/h/{{ pkgname }}/{{ pkgname }}-{{ version }}.tar.gz
+    # md5 is OK but sha256 is desirable.
+    md5: 08171310271823a83aff8269eeed2392
+
+build:
+    number: 0
+    # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Please drop this comment.

File: recipes/hiveplot/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+{% set maintainer = "ericmjl" %}
+
+package:
+    name: {{ pkgname }}
+    version: {{ version }}
+
+source:
+    fn: {{ pkgname }}-{{ version }}.tar.gz
+    url: https://pypi.io/packages/source/h/{{ pkgname }}/{{ pkgname }}-{{ version }}.tar.gz
+    # md5 is OK but sha256 is desirable.
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Please drop this comment.

File: recipes/hiveplot/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+# Jinja variables help maintain the recipe as you'll update the version only here.
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Please drop this comment.

File: recipes/hiveplot/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+{% set version = "0.1.8.4" %}
+{% set pkgname = "hiveplot" %}
+{% set maintainer = "ericmjl" %}
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Let's please drop this. It is really better to have a proper listing in the maintainers' section.

File: recipes/hiveplot/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        - icu
+    run:
+        - python
+        - matplotlib
+        - numpy
+        - pip
+        - icu
+
+test:
+    # Some package might need a `test/commands` key to check CLI.
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Please drop this comment.

File: recipes/hiveplot/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    imports:
+        - hiveplot
+
+about:
+    home: http://github.com/{{ maintainer }}/{{ pkgname }}
+    # Remember to specify the license variants for BSD, Apache, GPL, and LGLP.
+    # Prefer the short version, e.g: GPL-2.0 instead of GNU General Public License version 2.0
+    # See https://opensource.org/licenses/alphabetical
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Please drop this comment.

File: recipes/hiveplot/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    license: MIT
+    summary: 'Hive plots in Python!'
+
+extra:
+    recipe-maintainers:
+        # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Please drop this comment.

File: recipes/hiveplot/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+requirements:
+    build:
+        - python
+        # When setuptools is available add the `--single-version-externally-managed --record record.txt` above.
+        - setuptools
+        - matplotlib
+        - numpy
+        - pip
+        - icu
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Do we need all the requirements during the build stage? Or would `python`, `setuptools`, and `pip` be sufficient?

File: recipes/hiveplot/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+test:
+    # Some package might need a `test/commands` key to check CLI.
+    imports:
+        - hiveplot
+
+about:
+    home: http://github.com/{{ maintainer }}/{{ pkgname }}
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Could we please have the proper URL here sans jinja?

File: recipes/hiveplot/meta.yaml (hunk 1 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 2 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: That's right, and neither is it required for building. Have removed them both; they were a legacy from my automatic environment export.

Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 3 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: Legacy from my automatic environment export. Removed.

File: recipes/hiveplot/meta.yaml (hunk 4 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 5 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: Note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 6 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 7 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 8 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 9 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: Note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 10 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done

File: recipes/hiveplot/meta.yaml (hunk 11 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 13 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: Out of curiosity, will the jinja render properly?

Either way, I have removed it anyways.

note to self: done.

File: recipes/hiveplot/meta.yaml (hunk 12 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: I think that those three should be sufficient. Please pardon my ignorance, but in what circumstances would anything more than those 3 be necessary for the build stage? Would building Cython extensions be one case, in which Cython would be required then?

Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,38 @@
+{% set version = "0.1.8.4" %}
+{% set pkgname = "hiveplot" %}
+
+package:
+    name: {{ pkgname }}
+    version: {{ version }}
+
+source:
+    fn: {{ pkgname }}-{{ version }}.tar.gz
+    url: https://pypi.io/packages/source/h/{{ pkgname }}/{{ pkgname }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: Sorry missed this. Let's replace `/h/` here with `/{{ pkgname[0] }}/.

**Comment by jakirkham** [comment]
One last comment. Otherwise LGTM.

File: recipes/hiveplot/meta.yaml (hunk 14 above)
Comment (ericmjl) [inline recipes/hiveplot/meta.yaml]: note to self: done.

File: recipes/hiveplot/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,38 @@
+{% set version = "0.1.8.4" %}
+{% set pkgname = "hiveplot" %}
+
+package:
+    name: {{ pkgname }}
+    version: {{ version }}
+
+source:
+    fn: {{ pkgname }}-{{ version }}.tar.gz
+    url: https://pypi.io/packages/source/{{ pkgname[0] }}/{{ pkgname }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/hiveplot/meta.yaml]: So the full thing should be.

```
https://pypi.io/packages/source/{{ pkgname[0] }}/{{ pkgname }}/{{ pkgname }}-{{ version }}.tar.gz
```

**Comment by patricksnape** [comment]
URL is getting expanded incorrectly: `https://pypi.io/packages/source/h/hiveplot%EF%BC%8Fhiveplot-0.1.8.4.tar.gz`

File: recipes/hiveplot/meta.yaml [hunk 16]
```diff
@@ -0,0 +1,38 @@
+{% set version = "0.1.8.4" %}
+{% set pkgname = "hiveplot" %}
+
+package:
+    name: {{ pkgname }}
+    version: {{ version }}
+
+source:
+    fn: {{ pkgname }}-{{ version }}.tar.gz
+    url: https://pypi.io/packages/source/{{ pkgname[0] }}/{{ pkgname }}／{{ pkgname }}-{{ version }}.tar.gz
```
Comment (patricksnape) [inline recipes/hiveplot/meta.yaml]: That '/' after the first `{{ pkgname }}` is not a slash! It's a unicode fullwidth solidus apparently...

**Comment by ericmjl** [comment]
@patricksnape I'm sorry! I had Chinese text input enabled at that point, apparently... Should be all fixed now.

**Pull request merged**